doc/decisions/frontend_end_to_end_testing.md

```diff
@@ -0,0 +1,175 @@
+# E2E Testing
+
+as of 12.07.2023
+
+## Problem
+
+There is currently no existing e2e tests locally and in the pipeline.
+It lacks profound decision making which framework to pick and the frontend team needs to be d'accord with it.
+
+## Constraints
+
+1. [Behaviour-driven development](https://www.selenium.dev/documentation/test_practices/testing_types/#behavior-driven-development-bdd)
+2. [Page Object Design Pattern](https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/)
+3. Fast and easy prototyping of the tests
```

Comment:
:+1: but can you give more details what is needed that this is possible? Are there maybe react specific quirks?

Reply:
BDD and POM support is just required by the testing tools/libraries. BDD is a more difficult one, because the framework needs to parse the feature files, POM you can achieve anywhere where you have classes and objects.

Reply:
My comment actually only refers to "Fast and easy prototyping of the tests". So the "fast" and "easy" somehow should be better defined. What makes a framework "fast" or "easy"?

Reply:
Maybe thinking of Opensesame helps: we probably would use a different framework there (more likely Robot Framework). What are the differences in the constraints?

Reply:
I understand now, I will explain it a bit better.
I'm currently working on the alternative solution robot-framework. Soon I will push my thoughts.